packages/issuer-sdk-js/src/internal.ts

```diff
@@ -14,3 +16,31 @@ export function createActionInput(params: Record<string, any>): Utils.ActionInpu
   const prefixedObject = Object.fromEntries(prefixedEntries);
   return Utils.ActionInput.fromObject(prefixedObject);
 }
+
+export async function encrypt(
+  message: Uint8Array,
+  encryptionPublicKey: Uint8Array,
+  secretKey: Uint8Array,
+) {
+  const nonce = nacl.randomBytes(nacl.box.nonceLength);
+
+  const encrypted = nacl.box(message, nonce, encryptionPublicKey, secretKey);
+
+  if (encrypted === null)
+    throw Error(
+      `Couldn't encrypt. ${JSON.stringify(
+        {
+          message: base64.encode(message),
+          nonce: base64.encode(nonce),
+          receiverPublicKey: base64.encode(encryptionPublicKey),
+        },
+        null,
+        2,
+      )}`,
+    );
+
+  const fullMessage = new Uint8Array(nonce.length + encrypted.length);
+  fullMessage.set(nonce, 0);
+  fullMessage.set(encrypted, nonce.length);
+  return base64.encode(fullMessage);
```

Comment:
Feels wrong to settle on a representation here. I'd return the `Uint8Array` and let the caller decide if they want to `utf8.decode`, or if they want to `base64.encode`, etc.